### PR TITLE
Make `@Use` on feature and fixture method parallel

### PR DESCRIPTION
**Pull request by Vampire**
For `@Use` on a specification same-thread policy is necessary as the whole
spec execution is wrapped in a use-closure, so that also helper methods
can use the category methods.

For `@Use` on features or fixture method this is not necessary, as just the method invocation
is wrapped in the use-closure and this is then always run within one thread.

### CODE REVIEW

**Comment by Vampire** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1691** <a href="https://app.graphite.dev/github/pr/spockframework/spock/1691" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/spockframework/spock/1691?utm_source=stack-comment).

File: spock-specs/src/test/groovy/spock/util/mop/UseSpec.groovy [hunk 1]
```diff
@@ -83,6 +82,78 @@ class UseOnMethods extends Specification {
...
+  def "executing iterations in parallel works with annotated specification if enabled"() {
+    when:
+    getSpecificationContext()
+    def result = runner.runWithImports '''
+      @Use(StringExtensions)
+      class ASpec extends Specification {
+        @Shared
+        CountDownLatch latch = new CountDownLatch(2)
+
+        @IgnoreIf({ instance.specificationContext.currentSpec.childExecutionMode.orElse(null) == ExecutionMode.SAME_THREAD })
```
Comment (leonard84) [inline spock-specs/src/test/groovy/spock/util/mop/UseSpec.groovy:112]: Why is this necessary?

File: spock-specs/src/test/groovy/spock/util/mop/UseSpec.groovy [hunk 2]
```diff
@@ -83,6 +82,78 @@ class UseOnMethods extends Specification {
...
+          enabled true
+          fixed(4)
+        }
+      }
+    }
+  }
+
+  def "executing iterations in parallel works with annotated specification if enabled"() {
+    when:
+    getSpecificationContext()
```
[the remark this answers is not in the document]
Comment (Vampire) [reply spock-specs/src/test/groovy/spock/util/mop/UseSpec.groovy:105]: Tbh, I have no idea right now.
Maybe just a left-over.
The tests seem to work fine without them, so I removed them.

File: spock-specs/src/test/groovy/spock/util/mop/UseSpec.groovy [hunk 3]
```diff
@@ -83,6 +83,82 @@ class UseOnMethods extends Specification {
...
+          given:
+          latch.countDown()
+
+          expect:
+          !latch.await(10, TimeUnit.SECONDS)
+
+          where:
+          i << (1..2)
+        }
+      }
```
Comment (leonard84) [inline spock-specs/src/test/groovy/spock/util/mop/UseSpec.groovy:127]: Wouldn't it make more sense to refactor the test then to express the intent?
```suggestion
      @Use(StringExtensions)
      class ASpec extends Specification {
        def feature() {
          expect:
          specificationContext.currentSpec.childExecutionMode.orElse(null) == ExecutionMode.SAME_THREAD
        }
      }
```
